### DISCUSSION TITLE
Apple iOS app - Available on TestFlight

### DISCUSSION THREAD

**Discussion by JigSawFr** [astubenbord/paperless-mobile#204] [Show and tell]
Hi there,
A little thread to keep you informed about the iOS app 😀 

TestFlight link: https://testflight.apple.com/join/nmLJvzP9

Do hesitate to post your comments here.

Actually working to automate it with GitHub Actions.

Stay tuned.

**Comment by JigSawFr** [comment]
v2.3.5 published.

**Comment by astubenbord** [comment]
That's great to hear! Thanks for your efforts. 

How did you integrate the publishing process into CI? Did you use fastlane?

**Comment by JigSawFr** [comment]
> That's great to hear! Thanks for your efforts. 
> 
> 
> 
> How did you integrate the publishing process into CI? Did you use fastlane?

Np!
Yes, I’m using fastlane for deployment, configured and automated many things with var env.
Also using Ruby manager to fix version used on project, and cleaned the Xcode part. You can see some commits on my fork. But more to come.

Will do a PR with my edits, I’ve also created a mini README in iOS folder for this part ;)

**Reply by astubenbord** [reply]
Very cool! :) It should be possible for you to add the environment vars/secrets in this repository without me being able to see them right?

**Reply by JigSawFr** [reply]
Don’t think so, will need contributor role to add them, the only sensible part is the fact it’s using a private repository in my GH to clone certificates used on Apple Store linked to developper account. Will dig into it, but I’m trying to let you almost entire control I can do.
I’m exploring actions to auto rebase my fork and run CI on new tags to push new iOS release automatically on TestFlight and notify testers. Also need to reduce time used as it can be very expensive on GH to use macOS runner 😀

**Reply by astubenbord** [reply]
I see, thanks. Just to let you know: I don't manually tag releases. I trigger new releases manually from GH by running an action. So I don't think it is necessary to have the action listen on new tags if that is what you are trying to achieve :) 
Or do you mean you currently listen to new tags on this repository from your private repo? Then this would make sense of course.

**Reply by JigSawFr** [reply]
Yep, trying to listen to new tags on this repo to start my publish action (rebase fork, deploy)

**Comment by JigSawFr** [comment]
v2.3.8 published 🎉

**Reply by astubenbord** [reply]
Just tested it the first time on iOS and works very well, thanks for uploading! Do you know when it can be released to the App Store? Or will it stay on TestFlight?

**Reply by JigSawFr** [reply]
For the moment on TestFlight, I need to sort out GitHub CI to automate it and when we are confortable we can go on store ;)

**Comment by lagset** [comment]
Just some Feedback: Tried the Testflight-Version yesterday on an iPad Air. Two user environment, 600+ documents. It works super smooth!


Just stumbled over a view minor issues when using it on tablet screens:

- when opening the burger menu, the burger menu itself is overlayed by the navbar which happens to be on the left side on iPad/Tablets. This looks a bit off.

- swiping from the left edge to the middle, gets you one step back in many screens (e.g. in Tag details, correspondent details, document preview etc.) but not in the document details view.

Looking forward to the official app store release.

**Reply by lagset** [reply]
Just encountered a more relevant issue: Using the iOS Share Dialog with a PDF file and selecting paperless mobile as an app doesn't work.

There is something blurred, greenish looking launched in the background (animation) but it never gets in the foreground and disappears.

**Reply by JigSawFr** [reply]
> Just encountered a more relevant issue: Using the iOS Share Dialog with a PDF file and selecting paperless mobile as an app doesn't work.
> 
> There is something blurred, greenish looking launched in the background (animation) but it never gets in the foreground and disappears.

Confirmed for this part on iOS.
Need to dig into it.

**Comment by lagset** [comment]
JFYI:
Direct document download isn't working currently.

![IMG_20230715_174634_505.jpg](https://github.com/astubenbord/paperless-mobile/assets/15156011/6268da11-1fa9-4cb9-9856-effb09c2836d)

However, you can still save the file using the "open externally" oder "print" actions.

**Reply by astubenbord** [reply]
Probably a permissions problem, both open in app and in system viewer save to the temporary directory, while download saves to downloads/applications dir on iOS.

**Reply by astubenbord** [reply]
I'll look into it when I have time, I'll create a separate issue from your comment, thanks for reporting

**Comment by martenls** [comment]
Hey, just wanted to ask what the current status is with the iOS/TestFlight release as the latest TestFlight just expired a few days ago. Really loving the app so far 👍🏻

**Reply by JigSawFr** [reply]
Will push update, sorry was very busy at work, thanks for noticing.

**Reply by astubenbord** [reply]
@JigSawFr Hi, do you know if app ownership can be transferred? I'm thinking of getting into the developer program so I can publish the app and manage updates in the future.

**Comment by Philantrop** [comment]
TestFlight says currently no new testers are accepted. Is a regular release imminent or would it be possible to add slots?

**Comment by vduseev** [comment]
Hey Team, do you need any help publishing this app to App Store?

You have a perfect case for making the app paid only. There is a clear target group of users. The project is open source, so anyone can build it themselves and run directly on their iPhone if they don't want to pay a one-time fee.

Earnings from app can support further development. I would publish it as soon as possible and focus on support and further development. You have a clear path to ramen profitability.

@astubenbord @JigSawFr If I can help in any way (advise on incorporation if you consider it, taxes, app store optimization) feel free to reach out to me on Twitter (vduseev). I'd be happy to talk

**Reply by lagset** [reply]
JFYI: On my side, I would definitely pay a typical one time payment price, if this helps dealing with the costs of publishing this for iOS and iPadOS.

However, take your time, no pressure :)

**Reply by astubenbord** [reply]
Thanks for your offer @vduseev, I'll get in touch once the most important bugs are fixed! :)

While I really like the idea of fully free software (both in terms of "openness" and price), I also think that the yearly fee for publishing in the Apple AppStore is totally disproportionate, e.g. compared to the one-time payment of 25€ in the case of Google Play. I mean there are some monthly sponsors like yourself I just noticed (currently around 6$/month) which helps take some weight off my shoulders, but considering taxes it's currently just not enough to cover these costs. I'll definitely think about it. However, I also have the fear that having a price tag on the app might "scare off" some users from even downloading the app.

**Reply by beposec** [reply]
I understand that you are worried about scaring off other users. Especially because with swift-paperless a very good free alternative exists. I very much hope that the donations will be able to cover the costs for Apple in the future. For me i really look forward to have paperless-mobile on iOS :-)

**Comment by lagset** [comment]
Alternative App Stores coming with iOS 17.4 in march for EU users might be an alternative to distribute without expensive apple developer license, won't it?

https://www.apple.com/newsroom/2024/01/apple-announces-changes-to-ios-safari-and-the-app-store-in-the-european-union/

**Comment by beposec** [comment]
@astubenbord at least apple stopped the core technology fee and makes it possible to publish the free app in the AppStore without additional costs :-) 

https://developer.apple.com/news/
https://www.golem.de/news/core-technology-fee-apple-schafft-neue-gebuehr-fuer-ios-entwickler-teilweise-ab-2405-184813.html

**Comment by jmadden91** [comment]
Is this still available? I’m getting a “Beta not found” error
